Search/SearchManager.php

```diff
@@ -126,7 +126,7 @@ private function getMetadataForDocument(Document $document)
     /**
      * {@inheritdoc}
      */
-    public function deindex($object)
+    public function deindex($object, string $locale = null)
```

Comment:
It looks like if somebody extend it it still errors also when its optional so we need to go this way here with this way:

```php
public function deindex($object/*, string $locale = null*/)
{
     $locale = func_num_args() >= 2 ? func_get_arg(1) : null;
```

Reply:
Unfortunately phpstan still fails, if you call `deindex()` with 2 parameters. Do you have an idea how to fix that?

~~~
Method
Massive\Bundle\SearchBundle\Search\SearchManagerInterface::deindex()
invoked with 2 parameters, 1 required.
~~~

Reply:
Sure that error need to added to ignores. Not sure if there is any phpstan plugin or opitons to support things like `/*, string $locale = null*/`.

Reply:
If i would call the `deindex()` method directly on the SearchManager implementation, phpstan wouldn't fail, because it recognizes the `func_num_args()` in the code. But since we're calling the method on the interface, it can't check any code for `func_num_args()` and therefore will show the error

Reply:
But maybe its possible to fake it with phpdocs by adding on the class/interface `@method deIndex ...` phpdoc annotation

Reply:
@alexander-schranz Can we merge that for now?

Reply:
That the new optional parameter is correctly detected you would need to typehint in the code you use it the `SearchManagerInterface` as `SearchManager` e.g.:

```
	/**
	 * SearchManager
	 */
	private $searchManager;

	/**
	 * @param SearchManager $searchManager
	 */
    public function __construct(SearchManagerInterface $searchManager)
	{
		$this->searchManager = $searchManager;
	}

	public function someFunction(object $object): void
	{
		$this->searchManager->deindex($object, 'de');
	}
```

Reply:
yes 👍